### PR TITLE
Ensure low fee tickets are detected upon maturation.

### LINKED ISSUE #521: Low fee tickets isnt working
As reported by [@girino:decred.org on Matrix](https://matrix.to/#/!XijfkJigmGCmMkgTTl:decred.org/$156797243524684yfzhv:decred.org?via=decred.org&via=matrix.org), the low fee tickets feature is not working correctly.

Low fee tickets are showing in the users /tickets page, but not in the admin tickets page.

### PR DESCRIPTION
**Pull request by jholdstock**
The low fee tickets feature has been broken since inception.
There are multiple places where tickets are validated and
they had inconsistent error handling.

This PR also includes some logging and commenting I added
whilst investigating the issue.

Closes #521

### CODE REVIEW

File: backend/stakepoold/rpc/rpcserver/context.go [hunk 1]
```diff
@@ -552,30 +558,33 @@ func (ctx *AppContext) processNewTickets(nt NewTicketsForBlock) {
...
 	liveTicketsCount := len(ctx.LiveTicketsMSA)
 	ctx.Unlock()
-	log.Debug("processNewTickets ctx.Unlock")
+	log.Debug("processNewTickets: ctx.Unlock")
 
 	// Log ticket information outside of the handler.
 	go func() {
 		for ticket, msa := range newLiveTickets {
-			log.Infof("added new live ticket %v msa %v", ticket, msa)
+			log.Infof("processNewTickets: added new live ticket %v msa %v", ticket, msa)
```
Comment (JoeGruffins) [inline backend/stakepoold/rpc/rpcserver/context.go]: If this is info, the average person might not know what an msa is. Could we write it out? I see you do write it out later.

File: backend/stakepoold/rpc/rpcserver/context.go [hunk 2]
```diff
@@ -521,21 +526,22 @@ func (ctx *AppContext) processNewTickets(nt NewTicketsForBlock) {
 
 		msgTx, err := MsgTxFromHex(n.hex)
 		if err != nil {
-			log.Warnf("MsgTxFromHex failed for %v: %v", n.hex, err)
+			log.Warnf("processNewTickets: MsgTxFromHex failed for %v: %v", n.hex, err)
```
Comment (JoeGruffins) [inline backend/stakepoold/rpc/rpcserver/context.go]: If we end up adding an error package like suggested in #518 , we might also start using the convention in dcrwallet where the operation is declared at the beginning of a function:
const op = "processNewTickets"
Then when we make an error
err := fmt.Errorf("%v: everybody died", op)
Just a suggestion tho, we can always decide later when/if that issue gets worked on.

File: backend/stakepoold/rpc/rpcserver/context.go [hunk 3]
```diff
@@ -552,30 +558,33 @@ func (ctx *AppContext) processNewTickets(nt NewTicketsForBlock) {
...
 	// Log ticket information outside of the handler.
 	go func() {
 		for ticket, msa := range newLiveTickets {
-			log.Infof("added new live ticket %v msa %v", ticket, msa)
+			log.Infof("processNewTickets: added new live ticket %v msa %v", ticket, msa)
 		}
 
 		for ticket, msa := range newIgnoredLowFeeTickets {
-			log.Infof("added new ignored ticket %v msa %v", ticket, msa)
+			log.Infof("processNewTickets: added new ignored ticket %v msa %v", ticket, msa)
```
Comment (JoeGruffins) [inline backend/stakepoold/rpc/rpcserver/context.go]: msa here too

File: backend/stakepoold/rpc/rpcserver/context.go [hunk 4]
```diff
@@ -521,21 +526,22 @@ func (ctx *AppContext) processNewTickets(nt NewTicketsForBlock) {
...
+			newIgnoredLowFeeTickets[*n.ticket] = n.msa
 			log.Warnf("ignoring ticket %v for msa %v ticketFeesValid %v err %v",
 				n.ticket, n.msa, ticketFeesValid, err)
-			newIgnoredLowFeeTickets[*n.ticket] = n.msa
 		}
-
-		newLiveTickets[*n.ticket] = n.msa
 	}
 
-	log.Debug("processNewTickets ctx.Lock")
```
Comment (chappjc) [inline backend/stakepoold/rpc/rpcserver/context.go]: I think these lock/unlock debug logs can be removed

**Comment by JoeGruffins** [comment]
It works!

File: backend/stakepoold/rpc/rpcserver/context.go (hunk 2 above)
Comment (jholdstock) [reply backend/stakepoold/rpc/rpcserver/context.go]: We can go through and do this later in #518. I might actually make a start on that this afternoon

**Comment by jholdstock** [comment]
Thanks all for the reviews. I have implemented all suggestions apart from the error change from Joe, I will implement that in #518 

> after adding the tickets, they still show as invalid for the user. Do you see the same?

Yeah I see this as well. This was how the original implementation worked. We could possibly change this later, in another PR, but I don't think it is worth investing much more time into this feature. Having spoken with VSP admins, it seems like the feature is not really used since VSP staking was added to decrediton.

**Pull request merged**